### PR TITLE
Adding wiring pi

### PR DESCRIPTION
**Pull request by bseto**

### CODE REVIEW

**Comment by bseto** [comment]
Do not merge until the debug messages are removed. Will clean the code up on Saturday meeting.

File: SDL/include/WiringPiButtons.hpp [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+using namespace std::chrono;
+
+class WiringPiButtons
+{
+    public:
+      WiringPiButtons();
+      int getEvents();  
+
+    private:
+      enum Button { UP = 25, DOWN = 24, LEFT  = 23, RIGHT = 22, RELEASED = 0};
```
Comment (mpminardi) [inline SDL/include/WiringPiButtons.hpp]: Add a comment explaining the reasoning behind these numbers / the mapping to the physical pins.

File: SDL/include/WiringPiButtons.hpp [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+      void initButton(int buttonNumber);
+      Button State;
```
Comment (mpminardi) [inline SDL/include/WiringPiButtons.hpp]: Should "State" be capitalized?

File: SDL/include/WiringPiButtons.hpp [hunk 3]
```diff
@@ -0,0 +1,25 @@
+#ifndef WIRINGPIBUTTONS_HPP
+#define WIRINGPIBUTTONS_HPP
+
+#include <stdio.h> 
+#include <wiringPi.h>
+#include <chrono>
+
+using namespace std::chrono;
```
Comment (mpminardi) [inline SDL/include/WiringPiButtons.hpp]: I do not think we should get in the habit of using namespaces.

File: SDL/src/Main.cpp [hunk 4]
```diff
@@ -176,7 +175,33 @@ void processEvents()
         }
 }
 
-
+void processGPIO(int button)
+{
+    //enum Button { UP = 25, DOWN = 24, LEFT  = 23, RIGHT = 22, RELEASED = 0};
```
Comment (mpminardi) [inline SDL/src/Main.cpp]: We should be using the Button enum here. Also, remove this commented out code once we do that.

File: SDL/src/WiringPiButtons.cpp [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+    {
+        case RELEASED:
+            if (returnButton != 0)
+            {
+                int tempValue = returnButton;
+                returnButton = 0;
+	        printf("FUNCTION returned %d\n",tempValue);
+                return tempValue; 
+            }
+            if (!digitalRead(UP))       { State = UP; }
```
Comment (mpminardi) [inline SDL/src/WiringPiButtons.cpp]: Make these if statements look like your other if's, or maybe replace them with ternary operators

File: SDL/src/Main.cpp [hunk 6]
```diff
@@ -175,7 +166,31 @@ void processEvents()
...
+	    musicPlayer.nextSong();
+	    break;
+	case WiringPiButtons::VOLUMEUP:
+	    musicPlayer.changeVolume(0.1);
+	    break;
+	case WiringPiButtons::VOLUMEDOWN:
+	    musicPlayer.changeVolume(-0.1);
+	    break;
+	case WiringPiButtons::TOGGLEPLAY:
+	    printf("Space was pressed!\n");
```
Comment (mpminardi) [inline SDL/src/Main.cpp]: Remove this printf

**Pull request merged**